src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -95,5 +95,39 @@ public void MulticastOption_CreateSocketSetGetOption_GroupAndInterfaceIndex_SetS
                 Assert.Throws<SocketException>(() => socket.GetSocketOption(SocketOptionLevel.IP, SocketOptionName.AddMembership));
             }
         }
+
+        [Fact]
+        public void MulticastInterface_SetToInterfaceIndexAnySucceeds()
+        {
+            IPAddress ip = IPAddress.Parse ("239.255.255.250");
+            int index = 0;
+            using (Socket s = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                s.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(index));
+            }
+        }
+
+        [Fact]
+        public void MulticastInterface_SetToInterfaceIndexLoopbackSucceeds()
+        {
+            IPAddress ip = IPAddress.Parse ("239.255.255.250");
+            int index = 1;
+            using (Socket s = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                s.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(index));
+            }
+        }
```

Comment:
This test and the one above it can be condensed into a single [Theory], e.g.
```C#
[Theory]
[InlineData(0)]
[InlineData(1)]
public void MulticastInterface_SetToInterface_ValidIndex_Succeeds(int index)
{
    ...
}
```